soh/soh/Enhancements/debugger/actorViewer.cpp

```diff
@@ -0,0 +1,1261 @@
+        for (int i = 0; i < acMapping.size(); i++) {
+            if (ImGui::Selectable(acMapping[i].c_str())) {
+                category = i;
+                list = PopulateActorDropdown(category, list);
+                break;
+            }
+        }
+        ImGui::EndCombo();
+    }
+
+    if (ImGui::BeginCombo("Actor", filler.c_str())) {
+        if (gGlobalCtx != nullptr && lastSceneId != gGlobalCtx->sceneNum) {
+            list = PopulateActorDropdown(category, list);
+            lastSceneId = gGlobalCtx->sceneNum;
+        }
+        for (int i = 0; i < list.size(); i++) { 
+            std::string label = std::to_string(i) + ": " + list[i]->overlayEntry->name;
+            std::string description = GetActorDescription(list[i]->id);
+            if (description != "")
+                label += " (" + description + ")";
+
+            if (ImGui::Selectable(label.c_str())) {
+                rm = LIST;
+                display = *list[i];
+                actor = i;
+                filler = label;
+                break;
+            }
+        }
+        ImGui::EndCombo();
+    }
+
+    if (ImGui::TreeNode("Selected Actor")) {
+        dispOverlay = display.overlayEntry;
+        DrawGroupWithBorder([&]() {
+            ImGui::Text("Name: %s", dispOverlay != nullptr ? dispOverlay->name : "???");
+            ImGui::Text("Description: %s", GetActorDescription(display.id).c_str());
+            ImGui::Text("Category: %s", dispOverlay != nullptr ? acMapping[display.category].c_str() : "???");
+            ImGui::Text("ID: %d", display.id);
+            ImGui::Text("Parameters: %d", display.params);
+        });
+        
+        ImGui::PushItemWidth(ImGui::GetFontSize() * 6);
+
+        DrawGroupWithBorder([&]() {
+            ImGui::Text("Actor Position");
+            ImGui::InputScalar("x pos", ImGuiDataType_Float, &display.world.pos.x);
+            ImGui::SameLine();
+            ImGui::InputScalar("y pos", ImGuiDataType_Float, &display.world.pos.y);
+            ImGui::SameLine();
+            ImGui::InputScalar("z pos", ImGuiDataType_Float, &display.world.pos.z);
+        });        
+
+        DrawGroupWithBorder([&]() {
+            ImGui::Text("Actor Rotation");
+            ImGui::InputScalar("x rot", ImGuiDataType_S16, &display.world.rot.x);
+            ImGui::SameLine();
+            ImGui::InputScalar("y rot", ImGuiDataType_S16, &display.world.rot.y);
+            ImGui::SameLine();
+            ImGui::InputScalar("z rot", ImGuiDataType_S16, &display.world.rot.z);
+        });        
+
+        if (display.category == ACTORCAT_BOSS || display.category == ACTORCAT_ENEMY) {
+            ImGui::InputScalar("Enemy Health", ImGuiDataType_U8, &display.colChkInfo.health);
+            InsertHelpHoverText("Some actors might not use this!");
+        }
+
+        if (ImGui::Button("Kill")) {
+            Actor_Delete(&gGlobalCtx->actorCtx, &display, gGlobalCtx);
```

Comment:
This function isn't meant to be called outside of the actor engine.  `Actor_Kill` should be used here instead.